### PR TITLE
Upgrade testing framework

### LINKED ISSUE #368: Create human-graded expanded test case set from NBS base cases
Although the expander randomizer script successfully generates new test cases based on permuted values from the existing NBS cases, because match decisions are re-used, the cases aren't necessarily properly graded. Instead of using the random permutations generated, we want to create a larger set of more robust human training examples. The scope of this ticket is the following: for each of the 83 test cases in the original NBS test file, generate 2-4 permutations in which different combinations of fields are dropped (simulating missingness) or altered (simulating typos and entry quality issues), and then assign a match-grade based on a reasoned judgment of how an end-user would classify the result.

### PR DESCRIPTION
**Pull request by bamader**
## Description
This PR provides an overhaul and update to the Record Linker algorithm match performance testing suite. It provides a robust new source of testing data--combining regraded NBS cases with new cases designed to test feature missingness and data quality limitations, as well as corrects a deficiency in our existing seed file where the last digit of SSN was getting chopped off. The new included README summarizes the purpose of different testing files, but for more details on the actual development plan and interpretation of new testing results, see the associated document.

## Related Issues
Closes #368 

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: tests/algorithm/data/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# README
```
Comment (ericbuckley) [inline tests/algorithm/data/README.md:1]: Love this readme, thanks for adding it!

**Review by ericbuckley — APPROVED** [review]
great work
